### PR TITLE
Edit parameters of programming expressions

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Adds the ability to add, edit, and rearrange parameters of programming expressions.

`ParameterEditor` renders the basic form to edit parameters. `OrderableList` handles adding, rearranging, and removing parameters, and will be extended in a future PR to be reusable for examples.

Basic edit experience:
Screen Shot 2021-10-29 at 10 56 59 AM

Rearranging and removing (it's a bit jumpy, I'll look at improving that in a future PR):
https://user-images.githubusercontent.com/46464143/139482886-9d200a7f-59a2-4b65-92d0-95cf7d115ee8.mov

## Links

[figma mock](https://www.figma.com/proto/oJhM1JLs2vMaKvzcrFjZds/Block-Docs?page-id=0%3A1&node-id=102%3A2&viewport=241%2C48%2C0.36&scaling=min-zoom)

## Testing story

added unit tests for new component and new field being saved

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/lib/levelbuilder/code-docs-editor/OrderableList.jsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                className="fa fa-caret-down"
+              />
+            )}
+            <i
+              onClick={() => removeItem(idx)}
+              style={styles.controlButton}
+              className="fa fa-trash"
+            />
+          </div>
+          <ParameterEditor
```
Comment (tim-dot-org) [inline apps/src/lib/levelbuilder/code-docs-editor/OrderableList.jsx:56]: to make this component more generic we could extract this specific component by passing a render function as a param that takes the list index and renders the component.

File: apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx [hunk 2]
```diff
@@ -43,13 +47,22 @@ export default function ProgrammingExpressionEditor({
...
     }
     setIsSaving(true);
+    const copiedParameters = programmingExpression.parameters.map(p => {
+      const copied = {...p};
+      delete copied.key;
+      return copied;
+    });
+    const programmingExpressionToSave = {
+      ...programmingExpression,
+      parameters: JSON.stringify(copiedParameters)
```
Comment (tim-dot-org) [inline apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx]: does this need to be stringified? we're stringifying the surrounding expression already

**Review by tim-dot-org — APPROVED** [review]
just a couple of potential changes, but otherwise looks great! nice tests!

File: apps/src/lib/levelbuilder/code-docs-editor/OrderableList.jsx (hunk 1 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/code-docs-editor/OrderableList.jsx:56]: oo good idea. I wanted to use `elementType` as a prop type to pass in the actual component type but we need a higher version of `prop-types`.

I was planning on working through these things as part of a separate PR if that's okay.

Comment (tim-dot-org) [reply apps/src/lib/levelbuilder/code-docs-editor/OrderableList.jsx:56]: sounds good!

File: apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx (hunk 2 above)
Comment (bethanyaconnor) [reply apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx]: It has to do with the way parameters are permitted in Rails but the [docs](https://www.rubydoc.info/docs/rails/ActionController%2FParameters:permit) have some secret syntax I can use!

**Pull request merged**